newsfragments/970.doc.rst

```diff
@@ -0,0 +1 @@
+Clarified documentation around Nursery.start_soon and Nursery.start, that start_soon does not guarantee task ordering, and when you should use each.
```

Comment:
@njsmith - do we usually add newsfragments for documentation updates?

Reply:
@oremanj @njsmith Any opinions? I don't care whether we leave it in or take it out, but I'd like this pull request resolved